### ISSUE TITLE
fix: reject unknown reduction values in weighted_bce_loss

### ISSUE THREAD

**Issue by g8rdier** [g8rdier/hqnn-forge#192]
`weighted_bce_loss` handles `reduction` like this:

```python
if reduction == "mean":
    return weighted.mean()
elif reduction == "sum":
    return weighted.sum()
return weighted
```

Any string other than `"mean"` or `"sum"` (a typo such as `"Mean"` or `"avg"`) falls through and returns the per-sample tensor with no error. Then `loss.backward()` fails with "grad can be implicitly created only for scalar outputs", which is at least loud. Code that does `loss.item()` or logs the value gets a confusing error far from the cause. Code that calls `.mean()` itself hides the mistake entirely.

## Expected

Raise `ValueError` for anything outside `{"mean", "sum", "none"}`, the same as `F.binary_cross_entropy_with_logits` and `FocalLoss` do. Add a test for the rejection next to `TestWeightedBCELoss` in `tests/test_losses.py`.

This is a behaviour change, so it is kept out of the formatting PR #142. That PR adds the first numerical tests for this function.
